pkg/sdk/warehouse.go

```diff
@@ -0,0 +1,476 @@
+	MinClusterCount                 *uint8         `ddl:"parameter" db:"MIN_CLUSTER_COUNT"`
+	ScalingPolicy                   *ScalingPolicy `ddl:"parameter,single_quotes" db:"SCALING_POLICY"`
+	AutoSuspend                     *uint          `ddl:"parameter" db:"AUTO_SUSPEND"`
+	AutoResume                      *bool          `ddl:"parameter" db:"AUTO_RESUME"`
+	ResourceMonitor                 *string        `ddl:"parameter,double_quotes" db:"RESOURCE_MONITOR"`
+	Comment                         *string        `ddl:"parameter,single_quotes" db:"COMMENT"`
+	EnableQueryAcceleration         *bool          `ddl:"parameter" db:"ENABLE_QUERY_ACCELERATION"`
+	QueryAccelerationMaxScaleFactor *uint8         `ddl:"parameter" db:"QUERY_ACCELERATION_MAX_SCALE_FACTOR"`
+
+	// Object params
+	MaxConcurrencyLevel             *uint `ddl:"parameter" db:"MAX_CONCURRENCY_LEVEL"`
+	StatementQueuedTimeoutInSeconds *uint `ddl:"parameter" db:"STATEMENT_QUEUED_TIMEOUT_IN_SECONDS"`
+	StatementTimeoutInSeconds       *uint `ddl:"parameter" db:"STATEMENT_TIMEOUT_IN_SECONDS"`
+}
+
+type WarehouseUnsetField string
+
+const (
+	WarehouseTypeField                   WarehouseUnsetField = "WAREHOUSE_TYPE"
+	WarehouseSizeField                   WarehouseUnsetField = "WAREHOUSE_SIZE"
+	WaitForCompletionField               WarehouseUnsetField = "WAIT_FOR_COMPLETION"
+	MaxClusterCountField                 WarehouseUnsetField = "MAX_CLUSTER_COUNT"
+	MinClusterCountField                 WarehouseUnsetField = "MIN_CLUSTER_COUNT"
+	ScalingPolicyField                   WarehouseUnsetField = "SCALING_POLICY"
+	AutoSuspendField                     WarehouseUnsetField = "AUTO_SUSPEND"
+	AutoResumeField                      WarehouseUnsetField = "AUTO_RESUME"
+	ResourceMonitorField                 WarehouseUnsetField = "RESOURCE_MONITOR"
+	CommentField                         WarehouseUnsetField = "COMMENT"
+	EnableQueryAccelerationField         WarehouseUnsetField = "ENABLE_QUERY_ACCELERATION"
+	QueryAccelerationMaxScaleFactorField WarehouseUnsetField = "QUERY_ACCELERATION_MAX_SCALE_FACTOR"
+	MaxConcurrencyLevelField             WarehouseUnsetField = "MAX_CONCURRENCY_LEVEL"
+	StatementQueuedTimeoutInSecondsField WarehouseUnsetField = "STATEMENT_QUEUED_TIMEOUT_IN_SECONDS"
+	StatementTimeoutInSecondsField       WarehouseUnsetField = "STATEMENT_TIMEOUT_IN_SECONDS"
+)
+
+func (opts *WarehouseAlterOptions) validate() error {
+	if opts.name.FullyQualifiedName() == "" {
+		return fmt.Errorf("name must not be empty")
+	}
+
+	exclusivePointers := []interface{}{
+		opts.Suspend,
+		opts.Resume,
+		opts.AbortAllQueries,
+		opts.NewName,
+		opts.Set,
+		opts.Unset,
+		opts.SetTags,
+		opts.UnsetTags,
+	}
+	if err := checkExclusivePointers(exclusivePointers); err != nil {
+		return fmt.Errorf("exactly one of Suspend, Resume, AbortAllQueries, NewName, Set, Unset, SetTags and UnsetTags must be set: %w", err)
+	}
+
+	if opts.Suspend != nil && *opts.Suspend && opts.Resume != nil && *opts.Resume {
```

Comment:
checkEclsuviePointers is a cool function. we should do something similiar for checking that both are not true. maybe making a rules validation lib